tests/src/python/test_selective_masking.py

```diff
@@ -1095,13 +1191,130 @@ def test_layout_export_2_sources_masking(self):
+        self.assertTrue(layer)
+
+        map_settings = QgsMapSettings()
+        crs = QgsCoordinateReferenceSystem('epsg:4326')
+        extent = QgsRectangle(-1.0073971192118132, -0.7875782447946843, 0.87882587741257345, 0.51640826470600099)
+        map_settings.setBackgroundColor(QColor(152, 219, 249))
+        map_settings.setOutputSize(QSize(420, 280))
+        map_settings.setOutputDpi(72)
+        map_settings.setFlag(QgsMapSettings.Antialiasing, True)
+        map_settings.setFlag(QgsMapSettings.UseAdvancedEffects, False)
+        map_settings.setDestinationCrs(crs)
+        map_settings.setExtent(extent)
+
+        map_settings.setLayers([layer])
+
+        self.check_layout_export("layout_export_point_pattern_fill", 0, [layer], extent=extent)
+
+    def test_vector_centroid_fill(self):
+        """
+        Test masking when a centroid fill symbol layer is involved
+        """
+        self.assertTrue(QgsProject.instance().read(os.path.join(unitTestDataPath(), "selective_masking_fill_symbollayer.qgz")))
+
+        layer = QgsProject.instance().mapLayersByName('centroid_fill')[0]
+        self.assertTrue(layer)
+
+        map_settings = QgsMapSettings()
+        crs = QgsCoordinateReferenceSystem('epsg:4326')
+        extent = QgsRectangle(-1.0073971192118132, -0.7875782447946843, 0.87882587741257345, 0.51640826470600099)
+        map_settings.setBackgroundColor(QColor(152, 219, 249))
+        map_settings.setOutputSize(QSize(420, 280))
+        map_settings.setOutputDpi(72)
+        map_settings.setFlag(QgsMapSettings.Antialiasing, True)
+        map_settings.setFlag(QgsMapSettings.UseAdvancedEffects, False)
+        map_settings.setDestinationCrs(crs)
+
+        map_settings.setLayers([layer])
+
+        self.check_layout_export("layout_export_centroid_fill", 0, [layer], extent=extent)
+
+    def test_vector_random_generator_fill(self):
+        """
+        Test masking when a random generator fill symbol layer is involved
+        """
+        self.assertTrue(QgsProject.instance().read(os.path.join(unitTestDataPath(), "selective_masking_fill_symbollayer.qgz")))
+
+        layer = QgsProject.instance().mapLayersByName('random_generator_fill')[0]
+        self.assertTrue(layer)
+
+        map_settings = QgsMapSettings()
+        crs = QgsCoordinateReferenceSystem('epsg:4326')
+        extent = QgsRectangle(-1.0073971192118132, -0.7875782447946843, 0.87882587741257345, 0.51640826470600099)
+        map_settings.setBackgroundColor(QColor(152, 219, 249))
+        map_settings.setOutputSize(QSize(420, 280))
+        map_settings.setOutputDpi(72)
+        map_settings.setFlag(QgsMapSettings.Antialiasing, True)
+        map_settings.setFlag(QgsMapSettings.UseAdvancedEffects, False)
+        map_settings.setDestinationCrs(crs)
+
+        map_settings.setLayers([layer])
+
+        self.check_layout_export("layout_export_random_generator_fill", 0, [layer], extent=extent)
```

Comment:
Make sure these tests use the standard test font

Reply:
Done